test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc

```diff
@@ -112,6 +113,26 @@ TEST(QuicPlatformTest, QuicMapUtil) {
   EXPECT_FALSE(quic::QuicContainsValue(stdvec, 0));
 }
 
+TEST(QuicPlatformTest, QuicMockLog) {
+  ASSERT_EQ(quic::ERROR, quic::GetLogger().level());
+
+  {
+    // Test a mock log that is not capturing logs.
+    CREATE_QUIC_MOCK_LOG(log);
+    EXPECT_QUIC_LOG_CALL(log).Times(0);
+    QUIC_LOG(ERROR) << "This should be logged but not captured by the mock.";
+  }
+
+  {
+    // Test a mock log that captures logs.
+    CREATE_QUIC_MOCK_LOG(log);
+    log.StartCapturingLogs();
+
+    EXPECT_QUIC_LOG_CALL_CONTAINS(log, ERROR, "also captured");
+    QUIC_LOG(ERROR) << "This should be logged and also captured by the mock.";
+  }
```

Comment:
Test QUIC_LOG gets restored to original_sink

Reply:
Done.